src/Ilios/AuthenticationBundle/Voter/UserVoter.php

```diff
@@ -53,12 +53,13 @@ protected function isGranted($attribute, $requestedUser, $user = null)
             // 3. the current user has faculty/course director/developer role
             //    and has READ rights to the given user's primary school via the permissions system.
             case self::VIEW:
+                $allSchools = $requestedUser->getAllSchools();
                 return (
                     $user->getId() === $requestedUser->getId()
                     || (
                         $this->userHasRole($user, ['Course Director', 'Faculty', 'Developer'])
                         && (
-                            $this->schoolsAreIdentical($user->getSchool(), $requestedUser->getSchool())
+                            $allSchools->contains($user->getSchool())
```

Comment:
for performance reasons, i'd chain this
`$requestedUser->getAllSchools()->contains($user->getSchool())`

that way, schools only get loaded if previous checks are passing.